codemod/output/Evaporate.ts

```diff
@@ -0,0 +1,538 @@
+
+        if (ACTIVE_STATUSES.includes(file.status)) {
+          this._pause(file, force, promises);
+        }
+      }
+    }
+
+    return Promise.all(promises);
+  }
+
+  _pauseOne(id, force) {
+    const promises = [];
+    const file = this.pendingFiles[id];
+
+    if (typeof file === "undefined") {
+      promises.push(
+        Promise.reject("Cannot pause a file that has not been added.")
+      );
+    } else if (file.status === PAUSED) {
+      promises.push(
+        Promise.reject("Cannot pause a file that is already paused.")
+      );
+    }
+
+    if (!promises.length) {
+      this._pause(file, force, promises);
+    }
+
+    return Promise.all(promises);
+  }
+
+  _pause(fileUpload, force, promises) {
+    promises.push(fileUpload.pause(force));
+    removeAtIndex(this.filesInProcess, fileUpload);
+    removeAtIndex(this.queuedFiles, fileUpload);
+  }
+
+  resume(id) {
+    return typeof id === "undefined" ? this._resumeAll() : this._resumeOne(id);
+  }
+
+  _resumeAll() {
+    Global.l.d("Resuming all file uploads");
+
+    for (const key in this.pendingFiles) {
+      if (this.pendingFiles.hasOwnProperty(key)) {
+        const file = this.pendingFiles[key];
+
+        if (PAUSED_STATUSES.includes(file.status)) {
+          this.resumeFile(file);
+        }
+      }
+    }
+
+    return Promise.resolve();
+  }
+
+  _resumeOne(id) {
+    const file = this.pendingFiles[id];
+    const promises = [];
+
+    if (typeof file === "undefined") {
+      promises.push(Promise.reject("Cannot pause a file that does not exist."));
+    } else if (!PAUSED_STATUSES.includes(file.status)) {
+      promises.push(
+        Promise.reject("Cannot resume a file that has not been paused.")
+      );
+    } else {
+      this.resumeFile(file);
+    }
+
+    return Promise.all(promises);
+  }
+
+  resumeFile(fileUpload) {
+    fileUpload.resume();
+    this.queueFile(fileUpload);
+  }
+
+  forceRetry() {}
+
+  consumeRemainingSlots() {
+    let avail = this.config.maxConcurrentParts - this.evaporatingCount;
+
+    if (!avail) {
+      return;
+    }
+
+    for (let i = 0; i < this.filesInProcess.length; i++) {
+      const file = this.filesInProcess[i];
+      const consumed = file.consumeSlots();
+
+      if (consumed < 0) {
+        continue;
+      }
+
+      avail -= consumed;
+
+      if (!avail) {
+        return;
+      }
+    }
+  }
+
+  validateEvaporateOptions() {
+    this.supported = !(
+      typeof File === "undefined" || typeof Promise === "undefined"
+    );
+
+    if (!this.supported) {
+      return "Evaporate requires support for File and Promise";
+    }
+
+    if (this.config.readableStreams) {
+      if (typeof this.config.readableStreamPartMethod !== "function") {
+        return "Option readableStreamPartMethod is required when readableStreams is set.";
+      }
+    } else {
+      if (
+        typeof Blob === "undefined" ||
+        typeof (
+          Blob.prototype.webkitSlice ||
```

Comment:
Replace `Blob` with `(Blob as any)` would tackle the problem with the `Blob` extends you've done that's scoped only to this project.

Reply:
Done 😄